### PR TITLE
Feature gracefulrestart and defaultoriginate option

### PR DESCRIPTION
**Pull request by yktsubo**
This PL supports the following regarding Edge - OSPF routing options don't work #196  :
 - If bgp or ospf is enabled, gracefulrestart and defaultoriginate option will work.
 - If NSX 6.3.0 or later is used,  defaultoriginate option will return 'error'.

Current code needs 'EnableOSPF' or 'EnableBGP' option in order to change  gracefulrestart and defaultoriginate. ( # Honestly it can be good enough for me. but it's not intuitive)
I propose this fix however if current code is good enough for users I will cancel this PL.
Therefore please give me opinions.

BTW, I already tested this function with the my testcode which is not ready to commit.

### CODE REVIEW

**Comment by nmbradford** [comment]
gah - sorry Yuki - Id been sitting on partial fix for this for some time, you've done more than I did though.  There are tests for the OSPF portion visible at https://github.com/vmware/powernsx/compare/master...nmbradford:fix-196?expand=1

Can you add these tests and BGP related equivalent and add to this PR?

**Comment by yktsubo** [comment]
Oh, sorry. I didn't realize it.
Thank you for your comment. Yes, I will do next week.

File: tests/integration/03.LogicalRouter.Tests.ps1 [hunk 1]
```diff
@@ -252,6 +270,24 @@ Describe "Logical Routing" {
...
+            $rtg.bgp.gracefulRestart | should be false
+        }
+
+        it "Can enable Default Originate" {
+            $rtg = Get-NsxLogicalRouter $name | Get-NsxLogicalRouterRouting
+            $rtg | should not be $null
+            $rtg.bgp.defaultOriginate | should be false
+            $rtg | Set-NsxLogicalRouterBgp -DefaultOriginate -confirm:$false
+            $rtg = Get-NsxLogicalRouter $name | Get-NsxLogicalRouterRouting
+            $rtg.bgp.defaultOriginate | should be false
```
Comment (nmbradford) [inline tests/integration/03.LogicalRouter.Tests.ps1]: should be true?

**Comment by nmbradford** [comment]
Jenkins test this please

**Comment by nmbradford** [comment]
Yuki - as discussed, testing failed due to default originate not being supported on 6.3.x DLR and above.  As you have code to handle this, we just need to add a test that explicitly tests the failure scenario in 6.3.x

**Comment by nmbradford** [comment]
jenkins test this please

**Comment by nmbradford** [comment]
jenkins test this please

**Pull request merged**